cyberark/kubernetes-conjur-deploy
# Run Conjur cluster as a stateful set

This PR modifies Conjur master and standbys as a StatefulSet with persistent volumes in Kubernetes 1.5+ and OpenShift 3.5+. This makes it possible to restore the Conjur master manually when its pod is rescheduled.

Changes:
- Runs the Conjur master and standbys as a stateful set, with persistent volumes.
- A relaunch_master script is now available that terminates the master and reconfigures it. This script runs in Jenkins pipeline.
- Configures Conjur standbys and followers in parallel.
- haproxy now runs more properly, with default entrypoint and configuration reloading.

See Jenkins builds here: https://jenkins.conjur.net/view/k8s/job/cyberark--kubernetes-conjur-deploy/job/statefulset/

This PR replaces PR #7 .

To verify:
- Clone this branch
- Run demo scripts against GKE and OpenShift 3.7 and verify they work
